### DISCUSSION TITLE
Need update to "__dirname"

### DISCUSSION THREAD

**Discussion by grjan7** [koajs/koa#1636] [General]
The newer versions of Node.js does not support "__dirname".  It would be better if it can be replaced with "process.cwd()" or "path.resolve()".

This need to be made in the following file.

"/docs/guide.md"

**Comment by dougwilson** [comment]
Which version of Node.js? https://nodejs.org/dist/latest-v17.x/docs/api/modules.html#__dirname

**Comment by grjan7** [comment]
v14.16.0

**Reply by dougwilson** [reply]
https://nodejs.org/dist/latest-v14.x/docs/api/modules.html#modules_dirname

**Reply by dougwilson** [reply]
Seems to be in the Node.js 14 docs and I tried on 14.16 without issue. Are you getting an error? Have you reported your issue to the Node.js project?

**Comment by grjan7** [comment]
Yes, I do get the error. Module object does not have the property __dirname. I have not yet reported this issue to Node.js projects.

I receive the following error.
Uncaught ReferenceError: __dirname is not defined

**Comment by grjan7** [comment]
Thanks for your clarification.

It works fine when it is called from a file, but it remains inaccessible from console. 

Now, I have understood that the __dirname binds to the file where it is called from. Hence, it returns directory name only when they are called from file, otherwise it throws the reference error.

But, process.cwd() and path.resolve() returns the current working directory where the console is pointing at.

**Reply by dougwilson** [reply]
Ah, yes, in the REPL, `__dirname` is not available. But at the same time, `__dirname` is not the same as `process.cwd()`, as the prior is the directory in which the file was written is located and the latter is whatever the current working directory of the process is. When writing server code, basing paths off `__dirname` will ensure that no matter the cwd of the executing process, you are going to refer to the same files, especially when you upload the code to a hosting provider and start up your server as a daemon, etc.

**Comment by grjan7** [comment]
Yes, you are right. Initially, I misunderstood that both are same. Thank you very much for your timely answer that helped me correct my mistake.
